src/cdk/a11y/aria-reference.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/** IDs are deliminated by an empty space, as per the spec. */
+const ID_DELIMINATOR = ' ';
+
+/**
+ * Adds the given ID to the specified ARIA attribute on an element.
+ * Used for attributes such as aria-labelledby, aria-owns, etc.
+ */
+export function addAriaReferencedId(el: Element, attr: string, id: string) {
+  const ids = getAriaReferenceIds(el, attr);
+  if (ids.some(existingId => existingId.trim() == id.trim())) { return; }
+  ids.push(id.trim());
+
+  el.setAttribute(attr, ids.join(ID_DELIMINATOR));
+}
+
+/**
+ * Removes the given ID from the specified ARIA attribute on an element.
+ * Used for attributes such as aria-labelledby, aria-owns, etc.
+ */
+export function removeAriaReferencedId(el: Element, attr: string, id: string) {
+  const ids = getAriaReferenceIds(el, attr);
+  const filteredIds = ids.filter(val => val != id.trim());
```

Comment:
Looks like this needed to be revisited based on the regex, before the single space deliminator didn't trim up as much as I liked. The regex approach works much nicer